apps/uvc-app/uvc-gadget.cpp

```diff
@@ -1896,26 +1967,47 @@ int main(int argc, char *argv[]) {
     enum usb_device_speed speed = USB_SPEED_HIGH; /* High-Speed */
     enum io_method uvc_io_method = IO_METHOD_USERPTR;
 
-    std::vector<aditof::DeviceConstructionData> devsData;
     auto enumerator = aditof::DeviceEnumeratorFactory::buildDeviceEnumerator();
+    enumerator->findDevices(availableSensors);
 
-    enumerator->findDevices(devsData);
-
-    if (devsData.size() < 1) {
-        printf("No device was found!\n");
+    if (availableSensors.size() < 1) {
+        printf("No camera sensor is available!\n");
         return 1;
     }
+    const aditof::DeviceConstructionData &camSensorData =
+        availableSensors[sensorIndex];
+    std::string availableSensorsBlob;
+    // Serialize information about sensors to be send to the UVC client
+    for (const auto &eeprom : camSensorData.eeproms) {
+        availableSensorsBlob += "EEPROM_NAME=" + eeprom.driverName + ";";
+        availableSensorsBlob += "EEPROM_PATH=" + eeprom.driverPath + ";";
+    }
+    uint32_t buffLen = availableSensorsBlob.length();
```

Comment:
if `buffLen` is a read only variable, maybe make it `const`. From my point of view it increases the readability of the code. "oh, this is const, buffLen is a read only variable....";

Reply:
fixed